strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/domain/ArtifactIdGroup.java

```diff
@@ -0,0 +1,35 @@
+package org.carlspring.strongbox.domain;
+
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
+
+import javax.persistence.Entity;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Entity
+public class ArtifactIdGroup
+        extends ArtifactGroup
+{
+
+    public ArtifactIdGroup()
+    {
+        super();
+    }
+
+    public ArtifactIdGroup(final String name)
+    {
+        super(name);
+    }
+
+    public String getArtifactId()
+    {
+        return getName();
+    }
+
+    public static String createName(ArtifactCoordinates artifactCoordinates)
+    {
+        return String.format("%s/%s", artifactCoordinates.getClass().getSimpleName(), artifactCoordinates.getId());
```

Comment:
`artifactCoordinates.getClass()` can be DB proxy, so this can work not as expected

Reply:
We unproxy everything from the orientdb so it wouldn't happen in orientdb case. However, I agree there is some potential bug field here.

What do you think about adding and using here something like this:
```
.....
public interface ArtifactCoordinates  ....
{
     ....
    Class<T> getType();

}
```

and in maven case for example:
```
public class MavenArtifactCoordinates ....
{
    @Override
    public Class<MavenArtifactCoordinates> getType()
    {
        return MavenArtifactCoordinates.class;
    }
```
for security needs ?

Reply:
We already have `@ArtifactCoordinatesLayout` which defines layout name for every `ArtifactCoordinates` implementation.

Reply:
Changed to per repository solution.